### DISCUSSION TITLE
How to set data- attributes dynamically ?

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3464] [4. General]
Using [Flowbite's navbar](https://flowbite.com/blocks/application/sidenav/) & AlpineJS to dynamically populate the HTML and assign `:data-collapse-toggle="menu.id` to `:id="menu.id"` with an array of menu items.

```
The target element with id "customer-orders" does not exist. Please check the data-collapse-toggle attribute.
```

**Comment by ekwoka** [comment]
can you replicate? codepen is helpful. I'd expect it's just a race condition.

**Reply by anjanesh** [reply]
I solved this - sorry - `x-bind:data-collapse-toggle="menu.id"`

**Reply by ekwoka** [reply]
`x-bind:` and `:` are the same thing. and your error does not support the idea the data attribute was the one that was messed up.

**Reply by anjanesh** [reply]
Odd that it's working now with `:data-collapse-toggle="menu.id"` and wasn't yesterday.

Basically both these are getting loaded :
```
    <script src="https://cdnjs.cloudflare.com/ajax/libs/flowbite/1.6.4/flowbite.min.js"></script>
    <script src="https://unpkg.com/alpinejs@3.10.5/dist/cdn.min.js" defer></script>
```

**Reply by anjanesh** [reply]
I realize if I load alpinejs with `defer` then it doesn't work.

**Reply by SimoTod** [reply]
You need to make sure the other library starts after alpine. How do you start flowbite?

**Reply by anjanesh** [reply]
I have this `<script src="https://cdnjs.cloudflare.com/ajax/libs/flowbite/1.6.4/flowbite.min.js"></script>` before alpine now - will reverse it.

**Reply by SimoTod** [reply]
You'll also need to defer it otherwise it's going to kick in before Alpine anyway
